main.go

```diff
@@ -1,45 +1,67 @@
 package ofdconnector
 
 import (
-	"context"
 	"fmt"
 
-	"github.com/billz-2/ofd_connector/pkg/httpclient"
+	"github.com/billz-2/ofd_connector/internal/httpclient"
 )
 
 // OfdConnector interface defines the contract for OFD service operations
 type OfdConnector interface {
-	FiscalDriveList(context.Context) ([]FiscalDriveReaderInfo, error)
-	SetFactoryID(factoryID string)
-	// Add methods as needed
+	ZReport() ZReportI
+	Receipt() ReceiptI
 }
 
 type OfdConnectorConfigs struct {
 	ServiceAddress        string
 	RequestTimeOutSeconds int
+	FactoryID             string
 }
 
 // ofdConnector implements the OfdConnector interface
 type ofdConnector struct {
 	serviceAddress string
 	httpClient     httpclient.HTTPClient
 	factoryID      string
+	zReport        ZReportI
+	receipt        ReceiptI
 }
 
 // New creates a new instance of OfdConnector
 func New(configs OfdConnectorConfigs) (OfdConnector, error) {
 	if configs.ServiceAddress == "" {
 		return nil, fmt.Errorf("invalid url address")
 	}
+	if configs.FactoryID == "" {
+		return nil, fmt.Errorf("invalid FactoryID")
+	}
 
 	httpClient := httpclient.NewHTTPClient(configs.RequestTimeOutSeconds)
```

Comment:
If we are using HttpClient as a package - may be we could add ServiceAddress and FactoryID as a package's properties? not to pass them to all services we will be using, but to have them in HttpClient. we could also generate endpoint in HttpClient something like:
```
func (c httpClient)Endpoint(string Uri) string {
   return fmt.Sprintf("%s/%s/%s", c.ServiceAddress, Uri, c.FactoryID)
}
```
and use it like 
```
const receiptTxIdEndpoint = "FiscalDrive/Receipt/GetTXID"
//......
endpoint := r.httpClient.Endpoint(receiptTxIdEndpoint)
```